diff --git a/internal/hardhat-network/provider/provider.js b/internal/hardhat-network/provider/provider.js
index 5538259ab7fcbf99b7fa82e26922510da6444613..8d230d2421d977efc22f0dfdde676c6559420e88 100644
--- a/internal/hardhat-network/provider/provider.js
+++ b/internal/hardhat-network/provider/provider.js
@@ -415,7 +415,7 @@ async function createHardhatNetworkProvider(hardhatNetworkProviderConfig, logger
     log("Making tracing config");
     const tracingConfig = await makeTracingConfig(artifacts);
     log("Creating EDR provider");
-    const provider = EdrProviderWrapper.create(hardhatNetworkProviderConfig, loggerConfig, tracingConfig);
+    const provider = await EdrProviderWrapper.create(hardhatNetworkProviderConfig, loggerConfig, tracingConfig);
     log("EDR provider created");
     return provider;
 }
